internal/driver/driver_test.go

```diff
@@ -33,6 +38,147 @@ func TestDriver(t *testing.T) {
+			"$db", dbName,
+		))
+
+		body, err := wire.NewOpMsg(must.NotFail(doc.Encode()))
+		require.NoError(t, err)
+
+		_, _, err = c.Request(ctx, new(wire.MsgHeader), body)
+		require.NoError(t, err)
+	})
+
+	insertDocs := must.NotFail(bson.NewArray(
+		must.NotFail(bson.NewDocument("w", int32(2), "v", int32(1), "_id", int32(0))),
+		must.NotFail(bson.NewDocument("v", int32(2), "_id", int32(1))),
+		must.NotFail(bson.NewDocument("v", int32(3), "_id", int32(2))),
+	))
+
+	t.Run("Insert", func(t *testing.T) {
+		insertCmd := must.NotFail(bson.NewDocument(
+			"insert", "values",
+			"documents", insertDocs,
+			"ordered", true,
+			"lsid", must.NotFail(bson.NewDocument("id", lsid)),
+			"$db", dbName,
+		))
+
+		body, err := wire.NewOpMsg(must.NotFail(insertCmd.Encode()))
+		require.NoError(t, err)
+
+		resHeader, _, err := c.Request(ctx, new(wire.MsgHeader), body)
+		require.NoError(t, err)
+
+		assert.NotZero(t, resHeader.RequestID)
+	})
+
+	var cursorID int64
+
+	t.Run("Find", func(t *testing.T) {
+		findCmd := must.NotFail(bson.NewDocument(
+			"find", "values",
+			"filter", must.NotFail(bson.NewDocument()),
+			"sort", must.NotFail(bson.NewDocument("_id", int32(1))),
+			"lsid", must.NotFail(bson.NewDocument("id", lsid)),
+			"batchSize", int32(1),
+			"$db", dbName,
+		))
+
+		body, err := wire.NewOpMsg(must.NotFail(findCmd.Encode()))
+		require.NoError(t, err)
+
+		resHeader, resBody, err := c.Request(ctx, new(wire.MsgHeader), body)
+		require.NoError(t, err)
+		assert.NotZero(t, resHeader.RequestID)
+
+		resMsg, err := must.NotFail(resBody.(*wire.OpMsg).RawDocument()).Decode()
+		require.NoError(t, err)
+
+		cursor, err := resMsg.Get("cursor").(bson.RawDocument).Decode()
+		require.NoError(t, err)
+
+		firstBatch := cursor.Get("firstBatch").(bson.RawArray)
+		cursorID = cursor.Get("id").(int64)
+
+		expectedDocs := must.NotFail(bson.NewArray(
+			must.NotFail(bson.NewDocument("_id", int32(0), "w", int32(2), "v", int32(1))),
+		))
+
+		testutil.AssertEqual(t, must.NotFail(expectedDocs.Convert()), must.NotFail(firstBatch.Convert()))
+		require.NotZero(t, cursorID)
+	})
+
+	getMoreCmd := must.NotFail(bson.NewDocument(
+		"getMore", cursorID,
+		"collection", "values",
+		"lsid", must.NotFail(bson.NewDocument("id", lsid)),
+		"batchSize", int32(1),
+		"$db", dbName,
+	))
+
+	t.Run("GetMore", func(t *testing.T) {
+		for i := 0; i < 2; i++ {
+			body, err := wire.NewOpMsg(must.NotFail(getMoreCmd.Encode()))
+			require.NoError(t, err)
+
+			resHeader, resBody, err := c.Request(ctx, new(wire.MsgHeader), body)
+			require.NoError(t, err)
+			assert.NotZero(t, resHeader.RequestID)
+
+			resMsg, err := must.NotFail(resBody.(*wire.OpMsg).RawDocument()).Decode()
+			require.NoError(t, err)
+
+			cursor, err := resMsg.Get("cursor").(bson.RawDocument).Decode()
+			require.NoError(t, err)
+
+			nextBatch := cursor.Get("nextBatch").(bson.RawArray)
+			newCursorID := cursor.Get("id").(int64)
+
+			expectedDocs := must.NotFail(bson.NewArray(
```

Comment:
Let's define `expectedDocs` as a slice of expected values per iteration above as data, not as code. Otherwise, there is a dependency between `insertDocs` (constant data) and `expectedDocs` (data generated with code)

Reply:
What do you mean, should I put the `[]byte` as expected?

Reply:
Basically, `expectedDocs` should not depend on `i`. Instead

```go
expectedDocs := []bson.Array{...}

...

for i := 0; i < 2; i++ {
  testutil.AssertEqual(t, must.NotFail(expectedDocs[i].Convert()), must.NotFail(nextBatch.Convert()))
}
```